public/app/plugins/datasource/grafana-azure-monitor-datasource/utils/common.ts

```diff
@@ -6,6 +6,9 @@ import { AzureMonitorOption } from '../types';
 export const findOption = (options: AzureMonitorOption[], value: string | undefined) =>
   value ? options.find((v) => v.value === value) ?? { value, label: value } : null;
 
+export const findOptions = (options: AzureMonitorOption[], values: string[] | undefined) =>
+  values ? options.filter((option) => values.find((v) => v === option.value)) : [];
```

Comment:
This seems to have O(n^2) time complexity.

I'd rather optimize it with a map:

```js
export const findOptions = (options: AzureMonitorOption[], values: string[] = []) => {
  if (values.length === 0) {
    return [];
  }

  const set = values.reduce((accum, item) => {
    accum.add(item);
    
    return accum;
  }, new Set());

  return options.filter((option) => set.has(option.value));
}
```